### ISSUE TITLE
Code documentation.

### ISSUE THREAD

**Issue by Starciad** [Starciad/StardustDefender#18]
### Description

The project code does not have any documentation.

### Justification

Documentation is an important point for a concise understanding of how the code works, however this is not found here. It would be interesting to add xmlDocs to the project `StardustDefender.*.Core.csproj` and `StardustDefender.*.csproj` to increase the readability of the code, considering that in many places, the files are gigantic and without documentation, it obscures understanding from them.

### Main features

- Document projects `StardustDefender.*.Core.csproj` and `StardustDefender.*.csproj` with **xmlDocs**.

### Benefits & Potential Impact

With good documentation it is possible to understand how a code works and especially why certain designer questions were taken. This will not only help the main project maintainers, but also potential collaborators.

### Implementation

Just use the existing `docs` branch to document the code without interfering with the `development` branch.

**Comment by Starciad** [comment]
Documentation has been added to the project. 🤠
